src/pages/myproducts/myproducts.js

```diff
@@ -85,6 +89,33 @@ export const MyList = () => {
       });
   }, [debouncedSearch, selectedSubcategory]);
 
+    const handleDelete = () => {
```

Comment:
You should probably send the product name and expiry date to the delete method when you click the button so React knows which product to delete

Reply:
so the delete method would be something like `const handleDelete = (name, expiryDate) => {...}`

Reply:
and when you click the button you would do `onClick={() => handleDelete(p.name, p.expiryDate)}`. This way the the method will receive the product info which BE will delete